src/trace.ml

```diff
@@ -419,8 +421,12 @@ let log_new_loc s loc =
   end
 
 (** Flushing *)
+exception Pid_changed
 
 let flush_at s ~now =
+  (* If the PID has changed, then the process forked and we're in the subprocess.
+     Don't write anything to the file, and raise an exception to quit tracing *)
+  if s.pid <> s.getpid () then raise Pid_changed;
```

Comment:
I'm slightly sad that we call getpid at least 3 times for each flush. Could we change `put_ctf_header` to take the `pid` as a parameter instead of `getpid`?

Reply:
Oof, yes, we should definitely do that.